stackit/internal/services/loadbalancer/loadbalancer/resource.go

```diff
@@ -209,41 +216,65 @@ func (r *projectResource) Schema(_ context.Context, _ resource.SchemaRequest, re
 			"external_address": schema.StringAttribute{
 				Description: descriptions["external_address"],
 				Optional:    true,
-				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
 			},
 			"listeners": schema.ListNestedAttribute{
 				Description: descriptions["listeners"],
-				Optional:    true,
-				Computed:    true,
+				Required:    true,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.List{
+					listvalidator.SizeBetween(1, 20),
+				},
 				NestedObject: schema.NestedAttributeObject{
 					Attributes: map[string]schema.Attribute{
 						"display_name": schema.StringAttribute{
 							Description: descriptions["listeners.display_name"],
 							Optional:    true,
 							Computed:    true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
 						},
 						"port": schema.Int64Attribute{
 							Description: descriptions["port"],
 							Optional:    true,
 							Computed:    true,
+							PlanModifiers: []planmodifier.Int64{
+								int64planmodifier.RequiresReplace(),
+							},
 						},
 						"protocol": schema.StringAttribute{
 							Description: descriptions["protocol"],
 							Optional:    true,
 							Computed:    true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+							Validators: []validator.String{
```

Comment:
These oneOf string validators introduce maintenance effort for us, when the API supports new values. Are these defined as enum in the SDK or is this just documented in the OAS? 

Im not sure how we dealt with this in the other resources, but I think we usually dont have these validators

Reply:
True, but I did here since they are defined as an enum and the API error message doesn't indicate the possible values

Reply:
> Im not sure how we dealt with this in the other resources, but I think we usually dont have these validators

Currently we have `oneOf`s for the DNS and MongoDB Flex also

Reply:
I remember this from the onboarding of the LoadBalancer to the SDK. The way their OAS is defined, the generator doesnt catch that this is an enum and generates these fields as regular string data type. Ideally, the OAS should be changed so that in the SDK these fields have an enum data type, where the validation would be done, instead of having this in the provider schema. 

For now we can keep it, for better user-experience, but we should ask the LoadBalancer team to adapt their OAS

Reply:
Created a [story](https://dev.azure.com/schwarzit/schwarzit.cloudportal/_backlogs/backlog/Developer%20Tools/Epics/?workitem=543989) to investigate this issue and remove these from the TFP once it is fixed